iOS10-HW-11-Ivan-Rybkin/Sourses/ViewControllers/ViewController.swift

```diff
@@ -0,0 +1,271 @@
+        button.layer.cornerRadius = 18
+        button.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    private lazy var twitterButton: UIButton = {
+        let button = UIButton()
+        button.setTitle("Twitter", for: .normal)
+        button.setTitleColor(.white, for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 10, weight: .semibold)
+        button.applyShadow()
+        button.setImage(UIImage(named: "twitterImage" ), for: .normal)
+        button.imageEdgeInsets = UIEdgeInsets(top: 0, left: -45, bottom: 0, right: 0)
+        button.titleEdgeInsets = UIEdgeInsets(top: 0, left: -10, bottom: 0, right: 0)
+        button.layer.cornerRadius = 18
+        button.backgroundColor = UIColor(named: "twitterButtonColor")
+        button.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    private lazy var dontHaveLabel: UILabel = {
+        let label = UILabel()
+        label.text = "Dont have account?"
+        label.textColor = .lightGray
+        label.textAlignment = .center
+        label.font = UIFont.systemFont(ofSize: 12, weight: .medium)
+        return label
+    }()
+
+    private lazy var signUpButton: UIButton = {
+        let button = UIButton()
+        button.setTitle("Sign up", for: .normal)
+        button.setTitleColor(UIColor(named: "twitterButtonColor"), for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 12, weight: .medium)
+        button.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        return button
+    }()
+
+    private lazy var gradient: CAGradientLayer = {
+            let gradient = CAGradientLayer()
+            gradient.frame = view.bounds
+            gradient.colors = [UIColor.systemIndigo.cgColor, UIColor.white.cgColor]
+            gradient.startPoint = CGPoint(x: 0.5, y: 0)
+            gradient.endPoint = CGPoint(x: 0.5, y: 1)
+            return gradient
+        }()
+
+    // MARK: - Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.layer.insertSublayer(gradient, at: 0)
+        setupHierarchy()
+        setupLayout()
+    }
+
+    // MARK: - Setup
+
+    private func setupHierarchy() {
+        view.addSubview(loginLabel)
+        view.addSubview(nameTextField)
+        view.addSubview(passwordTextField)
+        view.addSubview(loginButton)
+        view.addSubview(forgotButton)
+        view.addSubview(connectLabel)
+        view.addSubview(stripeView)
+        view.addSubview(secondStripeView)
+        view.addSubview(facebookButton)
+        view.addSubview(twitterButton)
+        view.addSubview(dontHaveLabel)
+        view.addSubview(signUpButton)
```

Comment:
вот это лучше наверно так записать 
```suggestion
        [loginLabel, nameTextField, passwordTextField,
         loginButton, forgotButton, connectLabel,
         stripeView, secondStripeView, facebookButton,
         twitterButton, dontHaveLabel, signUpButton].forEach {
            view.addSubview($0)
        }
```